pkg/network/tracer/tracer_test.go

```diff
@@ -1013,15 +1013,16 @@ func testDNSStats(t *testing.T, tr *Tracer, domain string, success, failure, tim
 	queryMsg.SetQuestion(dns.Fqdn(domain), dns.TypeA)
 	queryMsg.RecursionDesired = true
 
-	dnsClient := new(dns.Client)
-	dnsConn, err := dnsClient.Dial(dnsServerAddr.String())
-	require.NoError(t, err)
-	defer dnsConn.Close()
-	dnsClientAddr := dnsConn.LocalAddr().(*net.UDPAddr)
+	var dnsClientAddr *net.UDPAddr
 	require.Eventually(t, func() bool {
+		dnsClient := new(dns.Client)
+		dnsConn, err := dnsClient.Dial(dnsServerAddr.String())
+		require.NoError(t, err)
+		dnsClientAddr = dnsConn.LocalAddr().(*net.UDPAddr)
 		_, _, err = dnsClient.ExchangeWithConn(queryMsg, dnsConn)
+		_ = dnsConn.Close()
 		return err == nil || timeout != 0
-	}, 3*time.Second, 100*time.Millisecond, "Failed to get dns response")
+	}, 6*time.Second, 100*time.Millisecond, "Failed to get dns response")
```

Comment:
why did this change?

Reply:
I did this to bring it in line with the same change I made on other DNS retries in tests... basically the client timeout is `2s`, so this gives us 3 potential tries